c7n/tags.py

```diff
@@ -524,3 +525,120 @@ def process_resource_set(self, resource_set, tags):
             Resources=[v[self.id_key] for v in resource_set],
             Tags=tags,
             DryRun=self.manager.config.dryrun)
+
+class NormalizeTag(Action):
+    """Transform the value of a tag.
+    Set the tag value to uppercase, title, lowercase. Strip text from a tag key.
+
+    .. code-block :: yaml
+
+        policies:
+          - name: ec2-service-transform-lower
+            resource: ec2
+            comment: |
+              ec2-service-tag-value-to-lower
+            query:
+              - instance-state-name: running
+            filters:
+              - "tag:testing8882": present
+            actions:
+              - type: normalize-tag
+                key: lower_key
+                action: lower
+
+          - name: ec2-service-strip
+            resource: ec2
+            comment: |
+              ec2-service-tag-strip-blah
+            query:
+              - instance-state-name: running
+            filters:
+              - "tag:testing8882": present
+            actions:
+              - type: normalize-tag
+                key: strip_key
+                action: strip
+                value: blah
+
+    """
+
+    schema = utils.type_schema(
+        'normalize-tag',
+        key={'type': 'string'},
+        action={'type': 'array', 'items': {
```

Comment:
afaics action is singleton from the enum not an array, it seems like the enum is taking precedence for the py jsonschema implementation, but would be good to call out the correct type per the docstring usage.

https://spacetelescope.github.io/understanding-json-schema/reference/generic.html#enumerated-values